java/memory/memory-core/src/main/java/org/apache/arrow/memory/BaseAllocator.java

```diff
@@ -187,7 +193,9 @@ void associateLedger(BufferLedger ledger) {
     assertOpen();
     if (DEBUG) {
       synchronized (DEBUG_LOCK) {
-        childLedgers.put(ledger, null);
+        if (childLedgers != null) {
```

Comment:
do we need this?

Reply:
To pass validation for dereference.of.nullable:

[dereference.of.nullable] dereference of possibly-null reference childLedgers